ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixParticipant.java

```diff
@@ -100,11 +101,18 @@ public void participate(List<AmbryStatsReport> ambryStatsReports, AccountStatsSt
   }
 
   @Override
-  public boolean setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
-    if (isSealed) {
-      sealedReplicas.add(replicaId);
-    } else {
-      sealedReplicas.remove(replicaId);
+  public boolean setReplicaSealedState(ReplicaId replicaId, ReplicaSealStatus replicaSealStatus) {
+    switch (replicaSealStatus) {
+      case SEALED:
+        sealedReplicas.add(replicaId);
+        partiallySealedReplicas.remove(replicaId);
+        break;
+      case PARTIALLY_SEALED:
+        partiallySealedReplicas.add(replicaId);
+        break;
+      case NOT_SEALED:
+        partiallySealedReplicas.remove(replicaId);
```

Comment:
Do we also need sealedReplica.remove(replicaId)?
It's not necessary we will go from not-sealed -> partial sealed-> sealed?
It's possible that we jump from not-sealed to sealed, right? Can we have test cases for it if we don't have them yet.

Reply:
fixed.

Reply:
Added tests too.